jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/RetrieveRegistryCredentialsStep.java

```diff
@@ -84,10 +84,8 @@ static RetrieveRegistryCredentialsStep forTargetImage(
         }
       }
 
-      // If no credentials found, give an info (not warning because in most cases, the base image is
-      // public and does not need extra credentials) and return empty.
       eventHandlers.dispatch(
-          LogEvent.info("No credentials could be retrieved for registry " + registry));
+          LogEvent.lifecycle("No credentials could be retrieved for registry " + registry));
```

Comment:
Does this mean this message will be printed every time for public repositories, or is the removed comment outdated?

Reply:
We could leave this as `info`. I just thought calling this out explicitly outweighs adding one more line to the output. It may help in the case where the user hits unauthorized and mistakenly thinks it's failing because of a wrong username or password while in reality Jib couldn't retrieve any credentials. For example, the user typed `-Djib.auth.password` instead of `-Djib.from.auth.password` or put `<auth><username>` outside `<from>`, and it will be easier and less frustrating to identify this mistake themselves. But if you think this change does more harm than good, I'm fine going back.

But to really answer your question, the line will get printed for public images on many repositories (e.g., Docker Hub for sure, and maybe either Quay, OpenShift, or both―I can't remember). But interestingly, GCR doesn't get to this line. This is because we first attempt pulling without credentials (i.e., pulling before calling this method) and GCR succeeds at the first attempt for public images.

As a side note, this first attempt without credentials may not necessarily be a good thing, considering that it is OK to send unrelated or wrong credentials when pulling public GCR images. It's then actually a bad thing for registries like Docker Hub for the reason I'll explain right below.

However, as I said, for registries like Docker Hub, unfortunately this line is always printed, as it requires going through the registry token auth flow for any image. (That is, the first attempt without credentials will always fail.) And when we do the token auth flow, our current code always retrieves and passes credentials, even for public images that don't require credentials. (Another side note: this can cause a problem like https://github.com/GoogleContainerTools/jib/issues/2050#issuecomment-553347560 for public images.)

Reply:
I vaguely remember somewhere that people were confused by this message for public repos. I could be wrong though.

Reply:
That's a fair point. I will leave the current log level. The issue I described can be handle by adding some line to our FAQ.